d3-series/materials/Week 3/Assignment/Viz3.html

```diff
@@ -0,0 +1,85 @@
+<html>
+
+    <head>
+        <title>TEST</title>
+    </head>
+
+    <body>
+        <div id="viz"></div>
+
+        <script src="https://d3js.org/d3.v4.min.js"></script>
+        <script>
+    
+            var height = 800;
+            var width = 1000;
+
+            var svg = d3.select("#viz")
+                .append('svg')
+                .attr('height', height)
+                .attr('width', width);
+
+            d3.json('data.json', function (data) {
+
+                updateViz(data);
+
+                var int = setInterval(function () {
+                    data.pop();
+                    updateViz(data);
+
+                    if (data.length === 0) {
+                        clearInterval(int);
+                    }
+                }, 2000);
+                });
+
+            function updateViz(data) {
+                var cx = 50;
+                var cy = 50;
+                var cir = svg.selectAll('circle')
+                    .data(data)
+                
+                cir.enter()
+                    .append('circle')
+                    .attr('cx',function(d,i){
+                        return Math.random()*1000;

```

Comment:
Since you have the height and width set at the top, you can reference them here as well to avoid the need to update multiple places in the code if the height/width values were to change.